### PR TITLE
feat: map view without navigation

### LINKED ISSUE #104: Add compatibility with `react-native-maps` for android.
Currently when using both the dependencies together we are getting `checkDebugDuplicateClasses error when having react-native-maps as dependency` error. We need to use `react-native-maps` to show static maps with custom marker using react-native-vector-icons` which is not supported by this sdk.

Below is the example of custom marker that we would like to implement with the dynamic text in the center:
Screenshot 2024-04-12 at 3 50 21 PM

Also their are other usecases that we are still not supported by this SDK. 

For more information refer to [Issue 76](https://github.com/googlemaps/react-native-navigation-sdk/issues/76)

### LINKED ISSUE #234: NavigationController affects all instances of the map
SDK version: 0.4.3

Using the `navigationController` to set destinations sets the destinations on all the instances of the map. This can easily be reproduces in the example app as well. 

```
const addMarkersToMap = useCallback(() => {
    mapViewController
      ?.isMyLocationEnabled()
      .then(() => {
        console.log('My Location Enabled');
      })
      .catch((e: any) => {
        console.log('My Location Disabled', e);
      });
    const addMarkerCalls = LatLngs.map(latLng => {
      return mapViewController?.addMarker({
        position: latLng,
        visible: true,
      });
    });

    Promise.all(addMarkerCalls).then(markers => {
      console.log('All markers added');
      navigationController.setDestinations(
        LatLngs.map(data => ({ position: data }))
      );

      const nonEmptyMarkers = markers.filter(Boolean) as Marker[];

      setAppointmentMarkers(nonEmptyMarkers);
    });
  }, [mapViewController, navigationController]);

  const onNavigationReady = useCallback(() => {
    console.log('onNavigationReady');

    addMarkersToMap();
  }, [addMarkersToMap]);
  ```
  
  This above code is in the `NavigationScreen` component in the example app. Our production app has something similar. Now when you navigate to the `MultipleMaps` screen you can see the destinations set in those maps too however you don't see the markers (because that is using mapViewController and not navigationController). 
  
 You can also reproduce this by setting a destination in map1 in the `MultipleMaps` screen and then noticing that the same destination is set in map2 as well. 
 
 We have two separate instances of the map in our application and we need them to work independently of each other without any interference between the two. 

Screenshot 2024-08-23 at 12 25 23 PM

Screenshot 2024-08-23 at 12 26 00 PM

### LINKED ISSUE #254: Add Support for Classic Google MapView as Separate Component
# Background
Currently, the package supports creating `NavigationView`, which displays Google Maps with navigation routes using the native Google Navigation SDK views. However, developers have expressed the need to display a classic Google Map without initiating navigation or showing existing routes. Since the Navigation SDK includes the Google Maps SDK, integrating other plugins for a classic `MapView` can lead to conflicts. Therefore, it would be beneficial for developers to have the ability to display a classic map within this package.

To address this, a separate `MapView` component should be implemented, allowing the classic map to be displayed without any effects from an ongoing or initialized navigation session. Although `NavigationView` can be used without navigation, it remains tied to the navigation state, preventing the display of a classic map once navigation is initialized. The new `MapView` component will bypass this limitation, sharing the same `MapController` as `NavigationView` but without showing any navigation-related information or being affected by changes in the navigation state.

# Implementation
## Android:

- A `MapFragment` should be used in place of `SupportNavigationFragment` when displaying the classic `MapView`.

- The native view manager for `NavigationView` should be updated to handle the creation of either `MapFragment` or `NavigationFragment`, based on whether a classic map or navigation is required. This will be done by sending a type flag through the `createFragment` method.

## iOS:

- On iOS, the same `GMSMapView` component can be reused for the classic map view. However, when using the new `MapView`, the navigation session will not be attached to the map, allowing it to function as a regular Google Map without showing navigation routes or being influenced by the navigation state. View type is controlled by sending a type flag through the `createFragment` method

These changes will ensure that both platforms handle the new `MapView` component independently from the navigation state, providing developers with the flexibility to use a classic map when needed.

# Workaround:
Until this new MapView component is implemented, developers can display a classic map view using a workaround, though it's not a perfect solution.

- On Android, the ongoing navigation session must be cleared, and the `NavigationViewController` method `setNavigationUIEnabled(false)` should be called. This disables the navigation UI and allows the map to function without showing navigation routes.

- On iOS, the `setNavigationUIEnabled(false)` method can be used without clearing the navigation session, as it hides the navigation route as well while keeping the session active, allowing the map to be displayed as a classic map.

> [!NOTE]  
> Since `NavigationView` and `MapView` will be separate components, transitioning between navigation and classic map views cannot be done seamlessly. If the view is intended to display navigation routes at any point, it is recommended to use NavigationView consistently to avoid interruptions in the navigation state.

### PR DESCRIPTION
**Pull request by illuminati1911**
Resolves #254
Resolves #104
Fixes #234

- Separate MapView component has been implemented, allowing the classic map to be displayed without any effects from an ongoing or initialized navigation session.
- If map without navigation is used
  - On Android a new fragment type `SupportMapFragment` is created instead of `SupportNavigationFragment`.
  - On iOS the same view will be used, but the navigation session will not be attached to the map
- Fixed map zooming on iOS. Previously it always zoomed into/out of user's location.
>
- [x] Tests pass
- [x] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: android/src/main/java/com/google/android/react/navsdk/MapViewFragment.java
Comment (jokerttu) [inline android/src/main/java/com/google/android/react/navsdk/MapViewFragment.java:1]: `MapViewFragment` has a lot of duplicate code with `NavViewFragment`
Following PR indroduces separate [MapViewController.java](https://github.com/googlemaps/react-native-navigation-sdk/pull/255/files#diff-1d026a9b98ba070ee96d8b2591553dded35011f5ba49bf8e773c830021c8cfdc) in PR
https://github.com/googlemaps/react-native-navigation-sdk/pull/255, that will move the shared methods to separate controller.

**Review by jokerttu — APPROVED** [review]
LGTM

File: android/src/main/java/com/google/android/react/navsdk/NavViewManager.java [hunk 1]
```diff
@@ -310,19 +327,31 @@ public Map<String, Object> getExportedCustomDirectEventTypeConstants() {
...
     setupLayout(root);
 
     FragmentActivity activity = (FragmentActivity) reactContext.getCurrentActivity();
     if (activity != null) {
       int viewId = root.getId();
-      NavViewFragment fragment = new NavViewFragment(reactContext, root.getId());
-      fragmentMap.put(viewId, new WeakReference<NavViewFragment>(fragment));
+      Fragment fragment;
+      // FragmentType 0 = MAP, 1 = NAVIGATION.
+      if (fragmentType == 0) {
```
Comment (ArturoSalazarB16) [inline android/src/main/java/com/google/android/react/navsdk/NavViewManager.java:338]: Nit: Please add a constant (or enum) we can reference here.

File: ios/react-native-navigation-sdk/NavView.m [hunk 2]
```diff
@@ -36,8 +36,11 @@ - (void)layoutSubviews {
   }
 }
 
-- (NavViewController *)initializeViewControllerWithStylingOptions:(NSDictionary *)stylingOptions {
+- (NavViewController *)initializeViewControllerWithStylingOptions:(NSDictionary *)stylingOptions
+                                                     fragmentType:(NSInteger)fragmentType {
   _viewController = [[NavViewController alloc] init];
+  // FragmentType 0 = MAP, 1 = NAVIGATION.
+  _viewController.isNavigationEnabled = fragmentType == 1;
```
Comment (ArturoSalazarB16) [inline ios/react-native-navigation-sdk/NavView.m:43]: Same as above. Let's please define constants for the values.

**Review by ArturoSalazarB16 — APPROVED** [review]

File: android/src/main/java/com/google/android/react/navsdk/NavViewManager.java (hunk 1 above)
Comment (illuminati1911) [reply android/src/main/java/com/google/android/react/navsdk/NavViewManager.java:338]: Yeah I think it's better that way. I decided to leave it out as it seemed there were no custom types used before in the native code side. Anyway I'll change this.

Comment (illuminati1911) [reply android/src/main/java/com/google/android/react/navsdk/NavViewManager.java]: Done.

File: ios/react-native-navigation-sdk/NavView.m (hunk 2 above)
Comment (illuminati1911) [reply ios/react-native-navigation-sdk/NavView.m]: Done.

**Pull request merged**